sycl/include/sycl/ext/oneapi/experimental/builtins.hpp

```diff
@@ -40,6 +49,30 @@ uint32_t to_uint32_t(sycl::marray<bfloat16, N> x, size_t start) {
 }
 } // namespace detail
 
+// The printf() function defined lower takes arguments as is and may need
+// a conversion to some printable type for proper output.
+// The specializations of printf_arument_type_traits replaing T with some
+// printable type P must do so only when T is convertible to P.
+// Otherwise, the behavior is undefined.
+template <typename T, typename SFINAE = void>
+struct printf_arument_type_traits {
+  using type = T;
+};
+
+// simd_view to one element needs conversion to underlying element type
```

Comment:
this should better go to esimd-specific header

Reply:
Hm, that is how I initially did it, then thought it is nice to see both the general class and the specialization in one place.
Ok, I can move this to esimd-specific file.

Reply:
The specialization is moved to esimd header in f9f89de31d7dbf3734ba1dc1a39785f88a2956d8